app/controllers/order_imports_controller.rb

```diff
@@ -44,51 +44,37 @@ def create_order_import!
     )
   end
 
-  def flash_import_result
-    case
-    when @import_result.blank?
-      flash_now(:notice, I18n.t('controllers.order_imports.create.blank'))
-    when @import_result.failed?
-      flash_now(:error, import_error_message)
-    else
-      flash_now(:notice, import_success_message)
-    end
-  end
-
-  def fail_on_error?
-    @order_import.try(:fail_on_error).present?
+  def import_exception_alert(exception)
+    Rails.logger.error "#{exception.message}\n#{exception.backtrace.join("\n")}"
+    set_flash(:error, import_exception_message(exception))
   end
 
-  def failure_msg_key
-    if fail_on_error?
-      'controllers.order_imports.create.fail_immediately'
-    else
-      'controllers.order_imports.create.fail_continue_on_error'
-    end
+  def import_exception_message(exception)
+    I18n.t('controllers.order_imports.create.error', error: exception.message)
   end
 
-  def import_error_message
-    I18n.t failure_msg_key, @import_result.to_h
+  def get_order_imports
+    @current_facility.order_imports.order("created_at DESC")
   end
 
-  def import_exception_alert(exception)
-    Rails.logger.error "#{exception.message}\n#{exception.backtrace.join("\n")}"
-    flash_now(:error, import_exception_message(exception))
+  def process_order_import!
+    raise "Please upload a valid import file" if upload_file.blank?
+    @order_import = create_order_import!
+    @order_import.process_upload!
+    deliver_report
   end
 
-  def import_exception_message(exception)
-    I18n.t('controllers.order_imports.create.error', error: exception.message)
+  def deliver_report
+    CsvReportMailer.delay.csv_report_email(report_recipient, report)
+    set_flash(:notice, t("controllers.order_imports.create.job_is_queued", email: @report_recipient))
   end
 
-  def import_success_message
-    I18n.t 'controllers.order_imports.create.success', @import_result.to_h
+  def report_recipient
+    @report_recipient ||= params[:report_recipient] ||= @order_import.creator.email
```

Comment:
The double `||=` feels really weird. Also, does this handle the empty string `params[:report_recipient]`? Would `@report_recipient ||= params[:report_recipient].presence || @order_import.creator.email` work?